pkg/apis/internal.admin.acorn.io/v1/resources_test.go

```diff
@@ -0,0 +1,264 @@
+package v1
+
+import (
+	"errors"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+func TestAdd(t *testing.T) {
+	// Define test cases
+	testCases := []struct {
+		name     string
+		current  Resources
+		incoming Resources
+		expected Resources
+	}{
+		{
+			name:    "add to empty resources",
+			current: Resources{},
+			incoming: Resources{
+				Counts:     Counts{Apps: 1},
+				Quantities: Quantities{VolumeStorage: resource.MustParse("1Mi")},
+			},
+			expected: Resources{
+				Counts:     Counts{Apps: 1},
+				Quantities: Quantities{VolumeStorage: resource.MustParse("1Mi")},
+			},
+		},
+		{
+			name: "add to existing resources",
+			current: Resources{
+				Counts:     Counts{Apps: 1},
+				Quantities: Quantities{VolumeStorage: resource.MustParse("1Mi")},
+			},
+			incoming: Resources{
+				Counts: Counts{
+					Apps:   1,
+					Images: 1,
+				},
+				Quantities: Quantities{
+					VolumeStorage: resource.MustParse("1Mi"),
+					CPU:           resource.MustParse("20m"),
+				},
+			},
+			expected: Resources{
+				Counts: Counts{
+					Apps:   2,
+					Images: 1,
+				},
+				Quantities: Quantities{
+					VolumeStorage: resource.MustParse("2Mi"),
+					CPU:           resource.MustParse("20m"),
+				},
+			},
+		},
+		{
+			name:    "does not change flags",
+			current: Resources{},
+			incoming: Resources{
+				Flags:      Flags{Unlimited: true},
+				Counts:     Counts{Apps: 1},
+				Quantities: Quantities{VolumeStorage: resource.MustParse("1Mi")},
+			},
+			expected: Resources{
+				Counts:     Counts{Apps: 1},
+				Quantities: Quantities{VolumeStorage: resource.MustParse("1Mi")},
+			},
+		},
+	}
+
+	// Run the test cases
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			current := NewResources()
+			current.Add(tc.current)
+
+			incoming := tc.incoming
+			current.Add(incoming)
+			current.Equals(tc.expected)
+		})
+	}
+}
+
+func TestRemove(t *testing.T) {
+	// Define test cases
+	testCases := []struct {
+		name     string
+		current  Resources
+		incoming Resources
+		all      bool
+		expected Resources
+	}{
+		{
+			name:    "remove from empty resources",
+			current: Resources{},
+			incoming: Resources{
+				Counts:     Counts{Apps: 1},
+				Quantities: Quantities{VolumeStorage: resource.MustParse("1Mi")},
+			},
+			expected: Resources{},
+		},
+		{
+			name: "remove from existing resources",
+			current: Resources{
+				Counts:     Counts{Apps: 1},
+				Quantities: Quantities{VolumeStorage: resource.MustParse("1Mi")},
+			},
+			incoming: Resources{
+				Counts:     Counts{Apps: 1},
+				Quantities: Quantities{VolumeStorage: resource.MustParse("1Mi")},
+			},
+			expected: Resources{},
+		},
+		{
+			name: "remove persistent counts with all",
```

Comment:
Can you add one more test after this one to make sure that PersistentCounts are not affected when `all` is false? Could help us prevent an accidental and nasty bug in the future.